src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -1424,10 +1425,14 @@ void ShenandoahFreeSet::find_regions_with_alloc_capacity(size_t &young_cset_regi
                 old_collector_leftmost, old_collector_rightmost, old_collector_leftmost_empty, old_collector_rightmost_empty,
                 old_collector_regions, old_collector_used);
 
-  _partitions.establish_mutator_intervals(mutator_leftmost, mutator_rightmost, mutator_leftmost_empty, mutator_rightmost_empty,
+  idx_t rightmost_idx = (mutator_leftmost == max_regions)? -1: (idx_t) mutator_rightmost;
```

Comment:
Does this also affect upstream? Should we have a non-genshen patch too?

Reply:
We should do this upstream as well.  (or it will go upstream when we merge, but that might take a while). I wanted to test and integrate "quickly" into our internal test distribution.

Reply:
See https://github.com/openjdk/jdk/pull/20974